web-api/src/mind.rs

```diff
@@ -200,6 +197,10 @@ impl DocumentProvider {
     fn get(&self, id: &DocumentId) -> Option<&Document> {
         self.documents.get(id)
     }
+
+    fn get_documents(&self) -> Vec<Document> {
```

Comment:
:art: getters are usually implemented via (mutable) references in rust, whenever heavy work like cloning is involved, the `into` (from owned)/`to` (from referenced) prefixes are more common.
```suggestion
    fn to_documents(&self) -> Vec<Document> {
```

Reply:
changed